server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1314,6 +1321,11 @@ private void parseSource(DefaultSearchContext context, SearchSourceBuilder sourc
             context.evaluateRequestShouldUseConcurrentSearch();
             return;
         }
+        // Can be marked false for majority cases for which star-tree cannot be used
+        // As we increment the cases where star-tree can be used, this can be set back to true
+        boolean canUseStarTree = this.indicesService.getCompositeIndexSettings().isStarTreeIndexCreationEnabled()
```

Comment:
it is a simple boolean check since indicesService is initialized and compositeIndexSettings too is initialized within it with the value already. 

We don't read the actual setting every time.

Reply:
I guess we have a consensus here to keep the logic in SearchService itself.

@jainankitk Do you have further comments on `this.indicesService.getCompositeIndexSettings().isStarTreeIndexCreationEnabled()`?